### PR TITLE
[WIP] PKG Install Support 

### PR DESCRIPTION
**Pull request by KorewaWatchful**
### This is an attempt at introducing .pkg support for vita3k 
# [TESTING NEEDED]
Concerning issues #533 and #669, a continuation of PR #588, this pull request aims at adding functionality for installing .pkg for games/dlc content. Works with game and dlc .pkg files. Big thanks to @motoharu-gosuto for psvpfstools, and @IllusionMan1212 for a lot of the code written here.
- [x] psvpfstools add
- [x] psvpfstools working
- [x] add pkg2zip and gui dialog from IllusionMan1212
- [x] sce-utils refactor from IllusionMan1212
- [x] drm decrypt
- [x] addcont support

### CODE REVIEW

**Comment by Zangetsu38** [comment]
hm juste one thinks, edit every time last commit for push fix, no create all time new commit

**Comment by KorewaWatchful** [comment]
Yeah I'm not very good at this...

**Comment by 1whatleytay** [comment]
We usually name our stuff like `module: what we did to the module`. So `external: add psvfstools` or `gui: add pkg decrypt menu`. Usually its good to structure it like this so we keep everything that's merged into master clean and easy to read. I will have to squash it otherwise though XD

**Comment by Zangetsu38** [comment]
ouch, this pr is very direty now, very need squash :/

**Comment by Zangetsu38** [comment]
what is this 
image
 ? rebase on master please ^^

File: .gitmodules [hunk 1]
```diff
@@ -9,7 +9,7 @@
 	url = https://github.com/aquynh/capstone.git
 [submodule "external/crypto-algorithms"]
 	path = external/crypto-algorithms
-	url = https://github.com/B-Con/crypto-algorithms
+	url = https://github.com/KorewaWatchful/crypto-algorithms
```
Comment (Zangetsu38) [inline .gitmodules:12]: wrong, need using vita3k git 🤔

Comment (KorewaWatchful) [reply .gitmodules:12]: not anymore :D

Comment (Zangetsu38) [reply .gitmodules:12]: rebase on master, you have uslkess commit almready pressent on master, don't have sense

Comment (ichisadashioko) [reply .gitmodules:12]: If we are about to change something that module, I think we should fork that repo to `Vita3K` and create a PR to that repo first.

**Comment by ichisadashioko** [comment]
There are more than 1000 files. If you add code from another module, you should add them as submodule. It is pretty hard to see what did you change.

**Comment by KorewaWatchful** [comment]
of course fam.

File: .gitmodules (hunk 1 above)
Comment (ichisadashioko) [reply .gitmodules:12]: Why do you fork that module to change `sha256_init` to `sha256_init_one`? Is there any naming collision?

Comment (KorewaWatchful) [reply .gitmodules:12]: yes there is

Comment (KorewaWatchful) [reply .gitmodules:12]: if you want to discuss, please go to discord. This is a draft PR as you can see.

File: vita3k/host/src/pkg.cpp [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+	for (const auto &file : fs::directory_iterator(host.title_id_src + "/sce_module/")) 
+	{
+        self2elf(file.path().string(), file.path().string() + "elf", SCE_KEYS, temp_klicensee);
+        fs::rename(file.path().string() + "elf", file.path().string());
+        make_fself(file.path().string(), file.path().string() + "fself");
+        fs::rename(file.path().string() + "fself", file.path().string());
+	}
+	
+	self2elf(host.title_id_src + "/eboot.bin", host.title_id_src + "/eboot.bin" + "elf", SCE_KEYS, temp_klicensee);
```
Comment (Zangetsu38) [inline vita3k/host/src/pkg.cpp:264]: not forget run format.bat ^^

**Comment by KorewaWatchful** [comment]
Mac OSX testers needed.

**Pull request closed without merging**